service/Microsoft.Azure.Devices/HttpRegistryManager.cs

```diff
@@ -1186,6 +1187,52 @@ public override Task<Twin> ReplaceTwinAsync(string deviceId, string moduleId, st
         }
 #endif
 
+#if ENABLE_MODULES_SDK
+        public override Task<E2EDiagnosticSetting> UpdateDiagnosticModuleSettingAsync(string deviceId, E2EDiagnosticSetting setting, string etag)
+        {
+            return this.UpdateDiagnosticModuleSettingAsync(deviceId, setting, etag, CancellationToken.None);
+        }
+
+        public override async Task<E2EDiagnosticSetting> UpdateDiagnosticModuleSettingAsync(string deviceId, E2EDiagnosticSetting setting, string etag, CancellationToken cancellationToken)
+        {
```

Comment:
need to check SamplingPercentage value range

Reply:
added in E2EDiagnosticSetting